File: cargo/bindings/cpp/src/buoy_station.cpp
#include "buoyfinder/buoy_station.hpp"
#include "buoyfinder.h"

namespace buoyfinder {

    BFBuoyStation::BFBuoyStation(BuoyStation* station) :
        m_ptr(station) {
    }

    BFBuoyStation::BFBuoyStation(const char* station_id, const char* name, double lat, double lon) :
        m_ptr(nullptr) {

		m_ptr = buoy_station_new(station_id, name, lat, lon);
    }

    BFBuoyStation::~BFBuoyStation() {
		buoy_station_free(m_ptr);
		m_ptr = nullptr;
    }

    const std::string BFBuoyStation::stationId() const {
		if (m_ptr == nullptr) {
			return "";
		}
		return buoy_station_id(m_ptr);
    }

    const std::string BFBuoyStation::name() const {
		if (m_ptr == nullptr) {
			return "";
		}
		return buoy_station_name(m_ptr);
    }

    const bool BFBuoyStation::isActive() const {
		if (m_ptr == nullptr) {
			return false;
		}
		return buoy_station_active(m_ptr);
    }
}

File: cargo/bindings/swift/BuoyStation.swift
//
//  BuoyStation.swift
//  BuoyFinder
//
//  Created by Matthew Iannucci on 8/28/18.
//  Copyright © 2018 Matthew Iannucci. All rights reserved.
//

import Foundation

class BuoyStation {
    
    private let raw: OpaquePointer
    
    init(_ raw: OpaquePointer) {
        self.raw = raw
    }
    
    init(stationId_: String, stationName: String, latitude: Double, longitude: Double) {
        raw = buoy_station_new(stationId_, stationName, latitude, longitude)
    }
    
    deinit {
        buoy_station_free(raw)
    }
    
    var stationId: String {
        get {
            return String(cString: buoy_station_id(raw))
        }
    }
    
    var name: String {
        get {
            return String(cString: buoy_station_name(raw))
        }
    }
    
    var isActive: Bool {
        get {
            return buoy_station_active(raw)
        }
    }
    
    
}
